src/pool/srv.c

```diff
@@ -72,11 +76,107 @@ fini(void)
 	return 0;
 }
 
+static int
+recreate_pool_layout()
+{
+	struct smd_pool_info    *pool_info = NULL, *tmp;
+	d_list_t                 pool_list;
+	char                     uuid_str[DAOS_UUID_STR_SIZE];
+	int			 rc = 0, pool_list_cnt, i, fd;
+	char			*path = NULL, *pool_dir = NULL;
+	uint64_t		 blob_sz;
+
+	D_INIT_LIST_HEAD(&pool_list);
+	rc = smd_pool_list(&pool_list, &pool_list_cnt);
```

Comment:
As we discussed offline, the pools in newborns/zombies will be mistakenly re-created as normal pools. We might need to adjust the pool creation/destroy logic to not rely on the pool files but rely on SMD for newborns/zombies cleanup?

Reply:
Taken care of in the new commit. I have added a new flag SMD_POOL_IN_CREATION in the smd table entry for SMD_DEV_TYPE_META. This flag will be set when the table entry for TABLE_POOLS[SMD_DEV_TYPE_META] is updated and cleared after the directory is renamed. This flag is checked again upon restart. If set it considers the pool as newborn. Tested the same using instrumentation.